### PR TITLE
Add remote register implementation.

### PR DESCRIPTION
**Pull request by archnode**
Fist implementation for automated registration via check.

### CODE REVIEW

File: src/Plugin/RemoteCheck/RemoteRegister.php [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+    $relaxed_root = $relaxed_config->get('api_root');
+    $relaxed_password = \Drupal::service('relaxed.sensitive_data.transformer')->get($relaxed_config->get('password'));
+
+    // We create an encoded uri for this site.
+    $uri = new Uri($site_host . $relaxed_root);
+    $uri = $uri->withUserInfo(
+      $relaxed_config->get('username'),
+      $relaxed_password
+    );
+    $encoded = \Drupal::service('relaxed.sensitive_data.transformer')->set((string) $uri);
```
Comment (fago) [inline src/Plugin/RemoteCheck/RemoteRegister.php:116]: needs to be dependency injected...

File: src/Plugin/RemoteCheck/RemoteRegister.php [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+    $body = [
+      'site_name' => $site_name,
+      'site_domain' => $site_host,
+      'site_uuid' => $site_uuid,
+    ];
+
+    // Additional information about the relaxed endpoint.
+    $relaxed_config = \Drupal::config('relaxed.settings');
+    $relaxed_root = $relaxed_config->get('api_root');
+    $relaxed_password = \Drupal::service('relaxed.sensitive_data.transformer')->get($relaxed_config->get('password'));
```
Comment (fago) [inline src/Plugin/RemoteCheck/RemoteRegister.php:108]: needs to be dependency injected...

File: src/Plugin/RemoteCheck/RemoteRegister.php [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+    $site_host = \Drupal::request()->getSchemeAndHttpHost();
+
+    $body = [
+      'site_name' => $site_name,
+      'site_domain' => $site_host,
+      'site_uuid' => $site_uuid,
+    ];
+
+    // Additional information about the relaxed endpoint.
+    $relaxed_config = \Drupal::config('relaxed.settings');
```
Comment (fago) [inline src/Plugin/RemoteCheck/RemoteRegister.php:106]: needs to be dependency injected...

File: src/Plugin/RemoteCheck/RemoteRegister.php [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+   * The serializer.
+   *
+   * @var \Symfony\Component\Serializer\Serializer
+   */
+  protected $serializer;
+
+  /**
+   * RemoteRegister constructor.
+   *
+   * @param array $configuration
```
Comment (fago) [inline src/Plugin/RemoteCheck/RemoteRegister.php:31]: Must have descriptions for every parameter.

File: src/Plugin/RemoteCheck/RemoteRegister.php [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+      watchdog_exception('relaxed', $e);
+    }
+  }
+
+  /**
+   * Collects the data necessary for registration on remote.
+   */
+  private function generateRegistrationPayload() {
+    // Basic Site information.
+    $config = \Drupal::config('system.site');
```
Comment (fago) [inline src/Plugin/RemoteCheck/RemoteRegister.php:94]: needs to be dependency injected...
you can use service-utils for many dependencies here - please add it.

**Review by fago — CHANGES_REQUESTED** [review]

File: src/Plugin/RemoteCheck/RemoteRegister.php (hunk 1 above)
Comment (archnode) [reply src/Plugin/RemoteCheck/RemoteRegister.php]: Already done in other branch

File: src/Plugin/RemoteCheck/RemoteRegister.php (hunk 2 above)
Comment (archnode) [reply src/Plugin/RemoteCheck/RemoteRegister.php]: Already done in other branch

**Comment by archnode** [comment]
This pr is superseded by #3

**Pull request closed without merging**